december.js

```diff
@@ -2716,459 +2703,714 @@ function checkEffects() {
+
+    static init() {
+        if (CustomTextTriggers.has_init) {
+            return;
+        }
+        CustomTextTriggers.has_init = false;
+
+        // Setup the effect lookup
+        CustomTextTriggers.effect_lookup = new Map();
+        for (let effect_cls of CustomTextTriggers.effects) {
+            effect_cls.init();
+            CustomTextTriggers.effect_lookup.set(effect_cls.command, 
+                {effect: effect_cls, handle: effect_cls.handleCommand});
+        }
+
+        // TODO make this "/effects arg" with some kind of handler?
+        // Add non-effect commands here
+        CustomTextTriggers.effect_lookup.set('/effects_disable', 
+            {effect: null, handle: CustomTextTriggers.disableEffects});
+        CustomTextTriggers.effect_lookup.set('/effects_enable', 
+            {effect: null, handle: CustomTextTriggers.enableEffects});
+        CustomTextTriggers.effect_lookup.set('/effects_stop', 
+            {effect: null, handle: CustomTextTriggers.stopEffects});
+
+        // testing
+        //CustomTextTriggers.effect_lookup.get('/padoru').handle('', []);
+
+    }
+
+    static isMod(username) {
+        try {
+            let is_mod = false;
+            $("#userlist").find('span[class$=userlist_owner],span[class$=userlist_siteadmin]').each(function() {
+                if ($(this).text() === username) {
+                    is_mod = true;
+                    return false;
+                }
+            });
+
+            return is_mod;
+        } catch (e) { return false; }
+    }
+
+    static isFirstMod() {
+        const first_mod_element = $("#userlist").find('span[class$=userlist_owner],span[class$=userlist_siteadmin]').first();
+        if (!first_mod_element) {
+            return false;
+        }
+
+        return first_mod_element.text() === CLIENT.name;
+    }
+
+    static randomElement(array) {
+        return array[Math.floor(Math.random() * array.length)];
+    }
+
+    /* Command handlers */
+    static handleChatMessage(msg_data) {
+        const is_shadowed = msg_data.username === CLIENT.name && msg_data.meta.shadow;
+        if (msg_data.username === "[server]" || is_shadowed) {
+            return;
+        }
+
+        if (!CustomTextTriggers.isMod(msg_data.username)) {
+            return;
+        }
+
+        // If this client was the one who sent the message
+        const did_send_the_message = CLIENT.name === msg_data.username;
+
+        const message_parts = msg_data.msg.trim().toLowerCase().replace(/\s\s+/igm, ' ').split(' ');
+        if (message_parts.length <= 0 || !message_parts[0]) {
+            return;
+        }
+
+        const [effect_name, ...effect_args] = message_parts;
+        if (effect_name[0] !== '/') {
+            return;
+        }
+
+        // Do the command if it exists
+        try {
+            CustomTextTriggers.effect_lookup.get(effect_name).handle(...effect_args, did_send_the_message);
```

Comment:
Since effect_args can be variable in length, we need to have required arguments at the front
```suggestion
            CustomTextTriggers.effect_lookup.get(effect_name).handle(did_send_the_message, ...effect_args);
```